sdks/java/io/iceberg/src/main/java/org/apache/beam/sdk/io/iceberg/WriteUngroupedRowsToFiles.java

```diff
@@ -265,14 +265,18 @@ public void finishBundle(FinishBundleContext c) throws Exception {
         return;
       }
       recordWriterManager.close();
-      for (Map.Entry<WindowedValue<IcebergDestination>, List<ManifestFile>> destinationAndFiles :
-          Preconditions.checkNotNull(recordWriterManager).getManifestFiles().entrySet()) {
+
+      for (Map.Entry<WindowedValue<IcebergDestination>, List<SerializableDataFile>>
+          destinationAndFiles :
+              Preconditions.checkNotNull(recordWriterManager)
+                  .getSerializableDataFiles()
+                  .entrySet()) {
         WindowedValue<IcebergDestination> windowedDestination = destinationAndFiles.getKey();
 
-        for (ManifestFile manifestFile : destinationAndFiles.getValue()) {
+        for (SerializableDataFile dataFile : destinationAndFiles.getValue()) {
           c.output(
               FileWriteResult.builder()
```

Comment:
What will happen if the workitem fails here. If data is already written to Iceberg, sounds like we'll never clean up such temporary data from failed bundles ?

Probably there's no correctness issue though if we never finalize data/manifest files related to such data written from failed bundles.

Reply:
That's true, failed bundles will likely leave behind unused files. 
But also true that these unused files will ultimately never be committed, so Iceberg will not consider them as part of the table. 

> Probably there's no correctness issue

In that sense, it's not as much of a problem as the one we face with FileIO or BigQueryIO. I think worst case is it will cause some noise and wasted space? If it is a legit problem, we can implement a similar pattern of 1) write to temp directory and 2) copy over.

Reply:
It might make sense to document this. I guess following might work as a workaround to cleanup such files.

https://iceberg.apache.org/docs/1.5.1/spark-procedures/#remove_orphan_files